mysticeti-core/src/syncer.rs

```diff
@@ -168,7 +198,7 @@ mod tests {
         DeliverBlock(Data<StatementBlock>),
     }
 
-    impl SimulatorState for Syncer<TestBlockHandler, bool, TestCommitObserver> {
+    impl SimulatorState for Syncer<TestBlockHandler, bool, Option<RoundNumber>, TestCommitObserver> {
```

Comment:
Just curious, why `SimulatorState` needs to be implemented? I thought simulation testing is done via swapping out the core thread.

Reply:
Yeah the actual simulator tests are somewhat different from the ones implemented for the syncer. My understanding here is that the `Simulator` is being used as a way to test the `Syncer` logic without having the need to spawn other components - ex the netsyc which implements the leader timeout logic. Thus we need to implement the `SimulatorState` for the `Syncer` .